CUJExecutor/buildProfile

```diff
@@ -3,75 +3,123 @@
 
 
 
-# Stashes a profile built based on the passed COMPILATION_SETTING.
-# which is a base profile, followed by a colon, and then a (possibly empty) sequence of compliation flagse
+# Stashes a profile built based on the passed COMPILATION_SETTING, which is a base profile (which could be a file location or one of the tokens {e - empty, u - cloud, s - startup}), followed by a colon, and then a (possibly empty) sequence of compliation flags
+# By default, if a prefix of a profile setting (e.g. e:pm is a prefix of e:pmcm) has a corresponding profile in the directory for passed CUJ, we will build on top of that profile (to save time and to decrease noise in comparisons between two settings if one is a prefix of the other). 
+# Further, during building, all prefix compilation settings will be stashed so that later calls may later use these intermediate profiles.
+# (e.g. if a call to this script is made on the takeout.txt CUJ and the compilation setting e:pmcm after after a call on the same CUJ and the compilation setting e:pm, then the profile for e:pmcm will be built on top of the one for e:pm and the prefix profile for e:pmc will also be saved to the corresponding directory)
+# 
+# if the -NC (or --no-caching) flag is provided, the profile will be built from scratch and further calls cannot build from prefix profiles generated by the call (although prefix profiles will be created and saved to the appropriate directory)
+# Further, the resulting profiles will have the sinature "-NC-" and the date. (This flag is intended to be useful for creating many instances of a profile and performing operations on them, such as intersecting them)
+#
+# If the passed base profile is a file location rather than a token, the profile will always be built from scratch, and the resulting profile will not be stashed (neither will any intermediate profiles),
```

Comment:
Even if the base profile exists?
    
    the profile will always be built from scratch